src/DittoMachine.sol

```diff
@@ -55,10 +55,21 @@ contract DittoMachine is ERC721, ERC721TokenReceiver, ERC1155TokenReceiver {
         uint256 term;
     }
 
+    //protoId is a precursor hash to a cloneId used to identify tokenId/erc20 pairs
+    mapping(uint256 => uint256) public protoIdToIndexHead;
+    // mapping to track the next index after the index head
+    mapping(uint256 => mapping(uint256 => uint256)) public protoIdToIndexToPrior;
+    // mapping to track the index before the index head
+    mapping(uint256 => mapping(uint256 => uint256)) public protoIdToIndexToAfter;
+
+    mapping(uint256 => uint256) public protoIdToDepth;
```

Comment:
can you add a descriptive comment for all the following photo variables?

Reply:
will do!